data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/index/DynamicIndexManager.java

```diff
@@ -100,5 +106,29 @@ private void setupIndexWithRetries(final IndexManager indexManager) throws IOExc
             }
         }
     }
+    // Restrictions on index names (https://opensearch.org/docs/1.0/opensearch/rest-api/create-index/#index-naming-restrictions)
+    private String normalizeIndex(final String indexName) {
```

Comment:
We should probably also check on the length. It isn't documented, but from looking at the code, I believe it is 255.

https://github.com/opensearch-project/OpenSearch/blob/2bd709bf201dde294e21d9959d951bb7f9e5e40a/server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java#L286C13-L288

https://github.com/opensearch-project/OpenSearch/blob/2bd709bf201dde294e21d9959d951bb7f9e5e40a/server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java#L150

Reply:
Do you think that's something we can add later? We would have to default to truncating right?

Reply:
I think we should add it before we release so that our normalization function remains consistent.